raiden/transfer/mediated_transfer/mediation_fee.py

```diff
@@ -57,18 +58,20 @@ def _collect_x_values(
     balance_in: Balance,
     balance_out: Balance,
     max_x: int,
-) -> List[float]:
+) -> List[Fraction]:
     """ Collect all relevant x values (edges of piece wise linear sections) """
     assert schedule_in._penalty_func
     assert schedule_out._penalty_func
     all_x_vals = [x - balance_in for x in schedule_in._penalty_func.x_list] + [
         balance_out - x for x in schedule_out._penalty_func.x_list
     ]
     limited_x_vals = (max(min(x, balance_out, max_x), 0) for x in all_x_vals)
-    return sorted(set(limited_x_vals))
+    return sorted({Fraction(x) for x in limited_x_vals})
```

Comment:
Why the curly braces here? I know it's the same as a set/dict but better be explicit. Can you change it back to `set()`?

Reply:
That would be a set constructor with a generator expression instead of a set comprehension. The latter is more efficient, but that won't really matter in this case, so I'm fine with either.

Reply:
Hmm I did not know of the efficiency difference here. Thanks for that info! But yeah I also think it does not matter at all here so please revert to set()